cmd/orb_test.go

```diff
@@ -3338,4 +3339,91 @@ Windows Server 2010
 		})
 
 	})
+
+	Describe("Orb diff", func() {
+		var (
+			token        string
+			tempSettings *clitest.TempSettings
+			command      *exec.Cmd
+		)
+
+		BeforeEach(func() {
+			token = "testtoken"
+			tempSettings = clitest.WithTempSettings()
+		})
+
+		AfterEach(func() {
+			tempSettings.Close()
+		})
+
+		DescribeTable("Shows the expected diff", func(source1, source2, expected, color string) {
+			orbName := "somenamespace/someorb"
+			version1 := "1.0.0"
+			orb1 := fmt.Sprintf("%s@%s", orbName, version1)
+			version2 := "2.0.0"
+			orb2 := fmt.Sprintf("%s@%s", orbName, version2)
+			command = exec.Command(pathCLI, "orb", "diff", orbName, version1, version2,
+				"--token", token,
+				"--host", tempSettings.TestServer.URL())
+
+			mockOrbSource(source1, orb1, token, tempSettings)
+			mockOrbSource(source2, orb2, token, tempSettings)
+
+			session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Eventually(session.Out).WithTimeout(5 * time.Second).Should(gbytes.Say(expected))
+			Eventually(session).Should(gexec.Exit(0))
+		},
+			Entry("Detect identical sources", "orb-source", "orb-source", "No diff found", "auto"),
+			Entry(
+				"Detect difference",
+				"line1\\nline3\\n",
+				"line1\\nline2\\n",
+				`--- somenamespace/someorb@1.0.0
+\+\+\+ somenamespace/someorb@2.0.0
+@@ -1,2 \+1,2 @@
+ line1
+-line3
+\+line2`,
+				"auto",
+			),
+		)
+	})
 })
+
+func mockOrbSource(source, orbVersion, token string, tempSettings *clitest.TempSettings) {
+	requestStruct := struct {
+		Query     string `json:"query"`
+		Variables struct {
+			OrbVersionRef string `json:"orbVersionRef"`
+		} `json:"variables"`
+	}{
+		Query: `query($orbVersionRef: String!) {
+			    orbVersion(orbVersionRef: $orbVersionRef) {
+			        id
+                                version
+                                orb { id }
+                                source
```

Comment:
Indentation issue ?
```suggestion
							version
							orb { id }
							source
```

Reply:
The indentation must be strictly the same as the one in the [call](https://github.com/CircleCI-Public/circleci-cli/blob/master/api/api.go#L1415)